compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ReachabilityAnalyzer.java

```diff
@@ -357,6 +357,8 @@ public void visit(BLangForkJoin forkJoin) {
 
     @Override
     public void visit(BLangMatchStatement matchStatement) {
+        checkStatementExecutionValidity(matchStatement);
```

Comment:
Shall we add another match test too (other than the change in the existing test)?